cg/services/flow_cell_service/flow_cell_service.py

```diff
@@ -0,0 +1,12 @@
+from cg.services.flow_cell_service.dtos import SequencingMetrics
+from cg.services.flow_cell_service.utils import create_metrics_dto
+from cg.store.store import Store
+
+
+class FlowCellService:
+    def __init__(self, store: Store):
+        self.store = store
+
+    def get_sequencing_metrics(self, flow_cell_name: str) -> list[SequencingMetrics]:
+        run = self.store.get_illumina_sequencing_run_by_device_internal_id(flow_cell_name)
+        return create_metrics_dto(run.sample_metrics) if run else []
```

Comment:
Are we not losing the "400 bad request" behaviour now? Which was previously returned when the flow cell was not found

Reply:
And error handling for when sequencing metrics are not found.

Reply:
Good point,that behavior is reasonable. The way it is written currently, the response would always be 200 but with an empty list. But it is better to be explicit about the resource not existing 👍

Reply:
We should likely adjust the CRUD to raise an error if something with an internal ID returns nothing and handle the error here.

Reply:
Yeah, I thought about that too! Kind of scary to change the behavior though because the method is called in a lot of places. I think I will consider that out of scope here

Reply:
Fixed in https://github.com/Clinical-Genomics/cg/pull/3549/commits/b793a370e72b6eaf8fceef6108bb62c1fb4ea3ae